src/pages/index.js

```diff
@@ -175,12 +205,87 @@ const editProfilePopup = new PopupWithForm(
 
 editProfilePopup.setEventListeners();
 
-/// UserInfo class ///
+const addPlacePopup = new PopupWithForm(
+  { popupSelector: "#profile__add-form" },
+  handleAddFormSubmit
+);
 
-const newUserInfo = new UserInfo({
-  // llamo a la clase para usar los metodos
-  nameSelector: "#profile__title", // los selectores que use, nose porque no llevan -input al final.
-  jobSelector: "#profile__description",
+addPlacePopup.setEventListeners();
+
+//// PopupWithImage class ////
+
+const popupImage = new PopupWithImage("#preview__image_modal"); //llamo a la clase (insatntiate)
+
+///submit functions  (add, edit, profilepicture )  ///
+
+function handleAddFormSubmit(inputValues) {
+  const { name, link } = inputValues; // Extract name and link from input values
+  addPlacePopup.setButtonText("Saving.."); // cree la funcion con el popUpWithForm y pase text como parametro
+  apiInstance
+    .addNewCard(name, link) // toma el name y el link
+    .then((newCard) => {
+      // newCard es la respuesta y contiene el name y link
+      console.log(newCard);
+
+      const cardElement = getCardView(newCard); // usamos esa respuesta y la ponemos en una constante junto con la funcion que genera la carta
+      cardSection.addItem(cardElement); // mandamos la carta recien creada con la info de API al browser o UI
+
+      addPlacePopup.close(); // cerramos el popup
+    })
+    .catch((err) => {
+      console.log("I got an error:", err.message);
+    })
+    .finally(() => {
+      addPlacePopup.setButtonText("Save");
+    });
+}
+
+function handleEditSubmit(inputValues) {
+  // una forma de resumir multiples parametros en uno solo
+  const { title, description } = inputValues; // toma los valores del "name" property en los inputs
+  editProfilePopup.setButtonText("Saving..");
+  // llamo el metodo usando el nombre de la constante que use para intantiate la clase
+  apiInstance
+    .editProfile(title, description) // metodo de api para obtener el user
+    .then((userInfo) => {
+      console.log(userInfo);
+      newUserInfo.setUserInfo(title, description);
+      editProfilePopup.close();
+    })
+    .catch((err) => console.error("I got an error:", err.message))
+    .finally(() => {
+      editProfilePopup.setButtonText("Save");
+    });
+
+  // llamo al close() usando la constante con la cual invoque la clase, imp: lo puedo hacer en cualquier lugar que requiera la funcion
+}
+
+function handleSubmitPicture() {
+  const input = document.querySelector("#profile__picture-input");
+  // inputValues = input.value;
+  const profilePicture = document.querySelector(".profile__image");
+  const newAvatarUrl = input.value.trim();
+  popupProfile.setButtonText("Saving..");
+  apiInstance
+    .updateProfile(newAvatarUrl)
+    .then((newAvatarUrl) => {
+      console.log(newAvatarUrl);
+      profilePicture.src = newAvatarUrl.avatar;
```

Comment:
Needs correction: It's needed to create a separate setter for avatar `setAvatar` and use it when needed to update the avatar

When initializing user info you can use bit methods:

```js
userInfo.setUserInfo({ title, description })
userInfo.setAvatar(avatar)
```